cluster/examples/kubernetes/ceph/cluster-test.yaml

```diff
@@ -39,3 +39,7 @@ spec:
     useAllNodes: true
     useAllDevices: true
     #deviceFilter:
+  # cleanupPolicy:
+  #   deleteDataDirOnHosts: "yes-really-destroy-data"
```

Comment:
I'm thinking we need to allow the `yes-really-destroy-data`, as part of the CR during creation. The scenario is common enough with dev/test environments that it seems worth it. But we still have to be very careful about allowing them to enable it in the CR. How about the following?
- If yes-really-destroy-data is specified in the CR:
   - If a configmap named `rook-uninstall-policy` exists and defines `yes-really-destroy-data: true`, then allow reconciling even with the destructive policy enabled
   - Else prevent reconciling
- Else don't destroy any data when the CR is deleted

This enhancement could be in a separate PR. It's really independent from this PR.